akka-docs/src/main/paradox/typed/actors.md

```diff
@@ -271,6 +271,149 @@ Therefore after creating the Actor system with the `main` Actor’s
+and mutable state is kept inside of it as fields. 
+
+Some reasons why you may want to do this are:
+
+* lambdas in Java can only close over final or effectively final fields, making it 
+  impractical to use this style in behaviors that mutate their fields
+* some state is not immutable, e.g. immutable collections are not widely used in Java
+* it could be more familiar and easier to migrate existing untyped actors to this style
+* mutable state can sometimes have better performance, e.g. mutable collections and 
+  avoiding allocating new instance for next behavior (be sure to benchmark if this is your 
+  motivation)
+
+### MutableBehavior API
+
+Defining a class based actor behavior starts with extending 
+@scala[`akka.actor.typed.scaladsl.MutableBehavior[T]`]
+@java[`akka.actor.typed.javadsl.MutableBehavior<T>`] where `T` is the type of messages 
+the behavior will accept.
+
+Let's repeat the chat room sample from @ref:[A more complex example above](#a-more-complex-example) but implemented
+using `MutableBehavior`. The protocol for interacting with the actor looks the same:
+
+Scala
+:  @@snip [MutableIntroSpec.scala](/akka-actor-typed-tests/src/test/scala/docs/akka/typed/MutableIntroSpec.scala) {  #chatroom-protocol }
+
+Java
+:  @@snip [MutableIntroTest.java](/akka-actor-typed-tests/src/test/java/jdocs/akka/typed/MutableIntroTest.java) {  #chatroom-protocol }
+
+Initially the client Actors only get access to an @scala[`ActorRef[GetSession]`]@java[`ActorRef<GetSession>`]
+which allows them to make the first step. Once a client’s session has been
+established it gets a `SessionGranted` message that contains a `handle` to
+unlock the next protocol step, posting messages. The `PostMessage`
+command will need to be sent to this particular address that represents the
+session that has been added to the chat room. The other aspect of a session is
+that the client has revealed its own address, via the `replyTo` argument, so that subsequent
+`MessagePosted` events can be sent to it.
+
+This illustrates how Actors can express more than just the equivalent of method
+calls on Java objects. The declared message types and their contents describe a
+full protocol that can involve multiple Actors and that can evolve over
+multiple steps. Here's the `MutableBehavior` implementation of the chat room protocol:
+
+Scala
+:  @@snip [MutableIntroSpec.scala](/akka-actor-typed-tests/src/test/scala/docs/akka/typed/MutableIntroSpec.scala) {  #chatroom-behavior }
+
+Java
+:  @@snip [MutableIntroTest.java](/akka-actor-typed-tests/src/test/java/jdocs/akka/typed/MutableIntroTest.java) {  #chatroom-behavior }
+
+The state is managed through fields in the class, just like with a regular object oriented class.
+As the state is mutable, we never return a different behavior from the message logic, but can return
+the `MutableBehavior` instance itself (`this`) as a behavior to use for processing the next message coming in.
```

Comment:
Perhaps mention that it's still possible to return a different `Behavior` and in that way different parts of the lifecycle can be implemented with `MutableBehavior` and others with the functional style.

Reply:
We have also a note in the style guide ticket that maybe we should recommend `Behaviors.same` over `this` in this case.